oioioi/contests/views.py

```diff
@@ -102,6 +106,49 @@ def get_contest_permissions(request, response):
     return response
 
 
+@menu_registry.register_decorator(
+    _("Rules"), lambda request: reverse('contest_rules'), order=100
+)
+@enforce_condition(contest_exists & can_enter_contest)
+def contest_rules_view(request):
+    no_of_rounds = get_number_of_rounds(request)
+    contest_dates = get_contest_dates(request)
+    submission_limit = get_problems_sumbmission_limit(request)
+    results_visibility = get_results_visibility(request)
+    scoring_type = request.contest.controller.description
```

Comment:
Personally, I’m not sure if `description` describes well the scoring type. For example `Polish Olympiad in Informatics - Onsite` doesn’t describe how the submission are scored and on sio2.mimuw we have even less meaningful descriptions.

Maybe a solution would be to have a field `scoring_description` on the contest controller and if this field is not present, the scoring type can be hidden.

Reply:
Good idea, implemented it in such a way. Whom I could contact so that I knwo what to write in `scoring_description` fields in different contest controllers?

Reply:
I'll write the descriptions now

Reply:
ProgrammingContestController
The submissions are scored on a set of groups of test cases. Each group is worth a certain number of points.
The score is a sum of the scores of all groups. The ranking is determined by the total score.
The full scoring is available after the results date for the round.

ACMContestController, ACMOpenContestController:
The solutions are judged on real-time.
The submission is correct if it passes all the test cases.
Participants are ranked by the number of solved problems. 
In case of a tie, the times of first correct submissions are summed up and a penalty of 20 minutes is added for each incorrect submission.
The lower the total time, the higher the rank.
Compilation errors and system errors are not considered as an incorrect submission.
The ranking is frozen 60 minutes before the end of the round.

AMPPZContestController:
same as ACMContestController, but the ranking is frozen 15 minutes before the end of the trial rounds and 60 minutes before the end of the normal rounds.

MPContestController:
The submissions are scored from 0 to 100 points. The participant can submit to finished rounds,
but a multiplier is applied to the score of such submissions.

OIContestController, OIOnsiteContestController:
The solutions are judged with sio2jail. They can be scored from 0 to 100 points.
If the submission runs for longer than half of the time limit, the points for this test are linearly decreased to 0.
The score for a group of test cases is the minimum score for any of the test cases.
The ranking is determined by the total score.
Until the end of the contest, participants can only see scoring of their submissions on example test cases.
Full scoring is available after the end of the contest.

OIFinalOnsiteContestController:
Same as OIContestController, but participants can reveal the full scoring of their submissions during the contest.

BOIOnsiteContestController, BOIOnlineContestController: not used

PAContestController:
The submissions are judged on real-time. All problems have 10 test groups, each worth 1 point.
If any of the tests in a group fails, the group is worth 0 points.
The ranking is determined by the total score.
The full scoring is available after the end of the round.
The ranking is determined by the total score and number of 10-score submissions, 9-score, 8...

PAFinalsContestController:
same as AMPPZContestController